_posts/mattox-endpoints.md

```diff
@@ -0,0 +1,184 @@
+---
+layout: post
+title: "Mattox, simple and pre-configured NServiceBus endpoints"
+author: Mauro Servienti
+synopsis: "NServiceBus endpoints support only code-based configuration, which is handy and not always friendly at the same time. Not all hope is lost, though. What if we could plugin the superb Microsoft configuration extension to configure NServiceBus endpoints?"
+header_image: /img/posts/mattox-endpoints/header.jpg
+tags:
+- nservicebus
+- messaging
+---
+
+[Matt Mattox](https://en.wikipedia.org/wiki/Matt_Mattox) was one of my favorite modern jazz ballet teachers. In the early 90s, I had the pleasure to spend a week at one of his workshops in Venice. One of the critical aspects of his dance style was isolation. I still remember this short exercise/choreography that finished with everyone screaming "Stop!" in a particular position. It was about isolating movements of body parts, stripping away all the gimmicks, and reducing it to its minimalist essence.
+
+[Mattox.NServiceBus](https://github.com/mauroservienti/Mattox.NServiceBus) is all about simplicity and NServiceBus endpoint configuration. With Mattox.NServiceBus, if we need to set up and start a new NServiceBus endpoint using Amazon SQS as the transport, we can do the following:
+
+```c#
+var endpoint = new AmazonSqsEndpoint("my-endpoint");
+var endpointInstance = await endpoint.Start();
+```
+
+If we're using Microsoft hosting, the above code will look like the following:
+
+```c#
+Host.CreateDefaultBuilder()
+    .UseNServiceBus(ctx => new AmazonSqsEndpoint(ctx.Configuration))
+    .Build();
+```
+
+At runtime, the [Mattox.NServiceBus.AmazonSQS](https://github.com/mauroservienti/Mattox.NServiceBus.AmazonSQS) endpoint will look for configuration settings through the provided [`IConfiguration` object instance](https://learn.microsoft.com/en-us/dotnet/core/extensions/configuration).
+
+For example, we can express the endpoint configuration using a json file similar to the  following:
+
+```json
+{
+  "NServiceBus": {
+    "EndpointConfiguration": {
+      "EndpointName": "my-endpoint",
+      "Auditing": {
+        "Enable": "true"
+      },
+      "Installers": {
+        "Enable": "true"
+      },
+      "Recoverability": {
+        "Delayed": {
+          "NumberOfRetries": 2,
+          "TimeIncrease": "00:00:25"
+        }
+      }
+    }
+  }
+}
+```
+
+With the above configuration, the endpoint name will be "my-endpoint," it'll have auditing enabled using the default "audit" queue, installers will configure the required infrastructure at runtime, and delayed retries will retry messages two times instead of the default three, with 25 seconds time increase.
```

Comment:
```suggestion
With the above configuration, the endpoint name will be "my-endpoint," it'll have auditing enabled using the default "audit" queue, installers will configure the required infrastructure at runtime, and delayed retries will retry messages two times instead of the default three, with a 25 second time increase between them.
```